InnerEye/ML/deep_learning_config.py

```diff
@@ -201,12 +201,29 @@ class WorkflowParams(param.Parameterized):
                                                           "associated with when performing k-fold cross validation")
     perform_training_set_inference: bool = \
         param.Boolean(False,
-                      doc="If True, run full image inference on the training set at the end of training. If False and "
-                          "perform_validation_and_test_set_inference is True (default), only run inference on "
-                          "validation and test set. If both flags are False do not run inference.")
+                      doc="If True, run full image inference on the training set at the end of training.")
     perform_validation_and_test_set_inference: bool = \
+        param.Boolean(None,
+                      doc="If True, run full image inference on validation and test sets after training.")
+    perform_validation_set_inference: bool = \
         param.Boolean(True,
-                      doc="If True (default), run full image inference on validation and test set after training.")
+                      doc="If True (default), run full image inference on validation set after training.")
+    perform_test_set_inference: bool = \
+        param.Boolean(True,
+                      doc="If True (default), run full image inference on test set after training.")
+    perform_ensemble_child_inference: bool = \
+        param.Boolean(None,
+                      doc="If True, run full image inference on the training, validation, and test sets at the end of ensemble "
+                          "child training.")
+    perform_ensemble_child_training_set_inference: bool = \
+        param.Boolean(False,
+                      doc="If True, run full image inference on the training set at the end of ensemble child training.")
+    perform_ensemble_child_validation_set_inference: bool = \
+        param.Boolean(False,
+                      doc="If True (default), run full image inference on validation set after ensemble child training.")
+    perform_ensemble_child_test_set_inference: bool = \
+        param.Boolean(False,
+                      doc="If True (default), run full image inference on test set after ensemble child training.")
```

Comment:
`perform_test_set_inference` is presently set up to change its semantics, depending on crossval or not. In my view, there should be `inference_on_test/val/test_set` that affects a normal run, and crossval child runs. And then 3 boolean flags `ensemble_inference_on_train/test/val` that only affect crossval ensemble builds. This would make your `perform_ensemble_child_inference` obsolete.

Reply:
Shruthi agrees and we've gone with option2.